src/mc.rs

```diff
@@ -1229,3 +1216,1069 @@ pub unsafe fn resize_rust<BD: BitDepth>(
+            sign: libc::c_int,
+            bitdepth_max: libc::c_int,
+        );
+    };
+
+    (blend, $name:ident) => {
+        pub(crate) fn $name(
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            tmp: *const DynPixel,
+            w: libc::c_int,
+            h: libc::c_int,
+            mask: *const uint8_t,
+        );
+    };
+
+    (blend_dir, $name:ident) => {
+        pub(crate) fn $name(
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            tmp: *const DynPixel,
+            w: libc::c_int,
+            h: libc::c_int,
+        );
+    };
+
+    (warp8x8, $name:ident) => {
+        pub(crate) fn $name(
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            src: *const DynPixel,
+            src_stride: ptrdiff_t,
+            abcd: *const int16_t,
+            mx: libc::c_int,
+            my: libc::c_int,
+            bitdepth_max: libc::c_int,
+        );
+    };
+
+    (warp8x8t, $name:ident) => {
+        pub(crate) fn $name(
+            tmp: *mut int16_t,
+            tmp_stride: ptrdiff_t,
+            src: *const DynPixel,
+            src_stride: ptrdiff_t,
+            abcd: *const int16_t,
+            mx: libc::c_int,
+            my: libc::c_int,
+            bitdepth_max: libc::c_int,
+        );
+    };
+
+    (emu_edge, $name:ident) => {
+        pub(crate) fn $name(
+            bw: intptr_t,
+            bh: intptr_t,
+            iw: intptr_t,
+            ih: intptr_t,
+            x: intptr_t,
+            y: intptr_t,
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            src: *const DynPixel,
+            src_stride: ptrdiff_t,
+        );
+    };
+
+    (resize, $name:ident) => {
+        pub(crate) fn $name(
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            src: *const DynPixel,
+            src_stride: ptrdiff_t,
+            dst_w: libc::c_int,
+            h: libc::c_int,
+            src_w: libc::c_int,
+            dx: libc::c_int,
+            mx: libc::c_int,
+            bitdepth_max: libc::c_int,
+        );
+    };
+}
+
+#[cfg(feature = "asm")]
+macro_rules! decl_fns {
+    ($mc_kind:ident, $name:ident) => {
+        decl_fns!($mc_kind, $name, sse2);
+        decl_fns!($mc_kind, $name, ssse3);
+
+        #[cfg(target_arch = "x86_64")]
+        decl_fns!($mc_kind, $name, avx2);
+
+        #[cfg(target_arch = "x86_64")]
+        decl_fns!($mc_kind, $name, avx512icl);
+    };
+
+    ($mc_kind:ident, $name:ident, $suffix:ident) => {
+        paste::paste! {
+            #[cfg(feature = "bitdepth_8")]
+            decl_fn!($mc_kind, [<$name _8bpc_ $suffix>]);
+            #[cfg(feature = "bitdepth_16")]
+            decl_fn!($mc_kind, [<$name _16bpc_ $suffix>]);
+        }
+    };
+}
+
+#[cfg(all(feature = "asm", any(target_arch = "x86", target_arch = "x86_64")))]
+#[allow(dead_code)] // Macro invocations generate more fn declarations than are actually used.
```

Comment:
Is it possible to avoid this?  It seems bad to massively over-declare functions that don't exist.

Reply:
If you base it on what happens in the `dsp_init` `fn`s, there shouldn't be any extra ones.

Reply:
Actually, ignore this for now.  Once the other little things are fixed, you can merge this, and we'll fix this later (since it may be simpler to do it along with #426 (see https://github.com/memorysafety/rav1d/pull/426/files#r1316922243).